src/systimer_posix.c

```diff
@@ -0,0 +1,102 @@
+/* Could live without sigaction but it seems to be pretty universal */
+
+/* This uses the POSIX 2001 API
+   which allows per-thread timing and minimises risk of
+   interference with other code using timers.
+
+   Sadly it's not always available, so we have an alternative implementation
+   below using the odler setitimer interface */
+
+/* Handler for the Alarm signal */
+
+int SyHaveAlarms = 1;
+
+/* This API lets us pick wich signal to use */
+#define TIMER_SIGNAL SIGVTALRM
+
+
+/* For now anyway we create one timer at initialisation and use it */
+static timer_t syTimer = 0;
+
+#if defined(SYS_IS_CYGWIN32) || defined(SYS_IS_FREEBSD)
```

Comment:
Checking for operating systems, instead of features, is generally frowned upon, because it doesn't scale well / is not future-proof.

At the very least, there should be a comment here explaining why cygwin and freebsd are treated differently here, so that people who need to modify this file in the future can understand it, and modify it if needed. Otherwise, in a few months or years, it will become a black block and a maintainance annoyance.